src/ResourceManager/ServiceBus/Commands.ServiceBus/Cmdlets/AuthorizationRule/SetAzureServiceBusAuthorizationRule.cs

```diff
@@ -0,0 +1,142 @@
+    public class SetAzureServiceBusAuthorizationRule : AzureServiceBusCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1, ParameterSetName = NamespaceAuthoRuleParameterSet,
+            HelpMessage = "Namespace Name.")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 1, ParameterSetName = QueueAuthoRuleParameterSet)]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 1, ParameterSetName = TopicAuthoRuleParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Namespace { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2, ParameterSetName = QueueAuthoRuleParameterSet,
+            HelpMessage = "Queue Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasQueueName)]
+        public string Queue { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2, ParameterSetName = TopicAuthoRuleParameterSet,
+            HelpMessage = "Topic Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasTopicName)]
+        public string Topic { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 3,
+            HelpMessage = "AuthorizationRule Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasAuthorizationRuleName)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 4,
+            ParameterSetName = AuthoRuleInputObjectParameterSet,
+            HelpMessage = "ServiceBus AuthorizationRule Object.")]
+        [Parameter(ParameterSetName = NamespaceAuthoRuleParameterSet)]
+        [Parameter(ParameterSetName = QueueAuthoRuleParameterSet)]
+        [Parameter(ParameterSetName = TopicAuthoRuleParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasAuthRuleObj)]
+        public SharedAccessAuthorizationRuleAttributes InputObject { get; set; }
+
+        [Parameter(Mandatory = true,
+            Position = 4,
+            ParameterSetName = AuthoRulePropertiesParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Rights, e.g.  @(\"Listen\",\"Send\",\"Manage\")")]
+        [Parameter(ParameterSetName = NamespaceAuthoRuleParameterSet)]
+        [Parameter(ParameterSetName = QueueAuthoRuleParameterSet)]
+        [Parameter(ParameterSetName = TopicAuthoRuleParameterSet)]
```

Comment:
@v-Ajnava make sure the parameters in this cmdlet have the same attributes across all parameter sets (might need to make sure that positions don't overlap)